frontend/cypress/integration/authRedirects.spec.js

```diff
@@ -180,14 +178,19 @@ describe('It validates `Add Project` behavior', () => {
     cy.reload()
   })
 
-  it('should redirect unauthenticated user to /login?redirect=/projects/new', () => {
+  it('should redirect unauthenticated user to /login and then back to /projects/new', () => {
     // The user is unauthenticated
-    cy.window().its('session.user').should('eq', null)
+    cy.window().its('session.user').should('not.ok')
 
     // Clicking `Add Project` redirects to the login page.
     // and adds redirect query to `/projects/new/`
     cy.get('#add_project').click()
-    cy.url().should('eq', 'http://kitspace.test:3000/login?redirect=/projects/new')
+
+    const username = getFakeUsername()
+    const email = faker.unique(faker.internet.email)
+    const password = '123456'
+    cy.createUser(username, email, password)
+    cy.url().should('eq', 'http://kitspace.test:3000/projects/new')
```

Comment:
Checking redirecting to `http://kitspace.test:3000/login?redirect=/projects/new` is deliberate. Because this one checks clicking on the `Add project` button. There's another test that checks redirecting to the page specified in the query param.
https://github.com/kitspace/kitspace-v2/blob/6fddd565bd2eecd7db2a2a12a15c9253362a41be/frontend/cypress/integration/authRedirects.spec.js#L78

Reply:
I'm not sure we should have e2e tests specific to the query parameter. With all of these tests I'm thinking about "what does the user care about in the moment". The user doesn't care if there's a `redirect` parameter or not. They care that they are redirected to the right place. Maybe we will change the query parameter at some point and use session storage to redirect. Should the test fail because we changed that?

I even think we should remove all the `cy.url().should('eq')` checks and replace them with `cy.get` for page-specific elements. (Of course, once we go public, we'd like to keep URLs consistent and can add checks for them then, but even then I'm not sure we should be checking a query parameter for the login redirect.)

Reply:
I try to test my assumptions about how it should behave, not just user interaction it makes changing any assumption doesn't go unnoticed.

Reply:
I'm not quite sure what you are saying, sorry. By "this makes changing.."  do you mean "this means that changing..."? What I'm just saying is the tests should emulate user behavior as much as possible and not rely on implementation details. You can still check your assumptions while not doing that.

Reply:
> I'm not quite sure what you are saying

> I try to test my assumptions about how it should behave

Anyways, we can test the implementation in unit tests.